udata/core/dataset/tasks.py

```diff
@@ -52,8 +52,7 @@ def purge_datasets(self):
         # Remove each dataset's resource's file
         storage = storages.resources
         for resource in dataset.resources:
-            if resource.fs_filename is not None:
-                storage.delete(resource.fs_filename)
+            dataset.remove_resource(resource)
```

Comment:
Iterating on a list while removing its elem seem like a bad idea (skip some iteration) sadly.

Reply:
Put back the logic:
```
if resource.fs_filename is not None:
    storage.delete(resource.fs_filename)
Dataset.on_resource_removed.send(Dataset, document=dataset, resource_id=resource.id)
```
without the `dataset.remove_resource`, as the resources will be deleted with the dataset deletion and the important is to delete the file in the storage and to emit the signal.